src/Orchard.Web/Core/Settings/Descriptor/ShellDescriptorManager.cs

```diff
@@ -51,7 +51,10 @@ public class ShellDescriptorManager : Component, IShellDescriptorManager {
         }
 
         private ShellDescriptorRecord GetDescriptorRecord() {
-            return _shellDescriptorRepository.Get(x => x != null);
+            if (_shellDescriptorRecord == null) {
```

Comment:
Any call to Update should reset the private field then.

Reply:
I thought it wasn't necessary because UpdateShellDescriptor gets the ShellDescriptorRecord from GetDescriptorRecord, so I supposed every operation on it affects the private property too. I did test enabling and disabling features and it seems to work as expected.